File: series2/series2_code/series2_code/ssprk3_timestep/plot_convergence.py
from numpy import *
import matplotlib
import matplotlib.pyplot as plt

error = loadtxt('error.txt')
dt = loadtxt('dt.txt')

# We ignore the first few datapoints,
# as we can see from the plot they are a bit wrong
rate = polyfit(log(dt[2:]), log(error[2:]), 1)[0]
print(rate)

plt.loglog(dt, error, '-o', label='$|u(T)-U_N|$')
plt.loglog(dt, error[-1]*dt**rate/(dt[-1]**rate), '--', label='$\Delta t^{%0.2f}$' % rate)
plt.xlabel('$\Delta t$')
plt.ylabel('$Error$')
plt.xlim([0.8*dt[-1], 1.2*dt[0]])
plt.legend(loc=0)
plt.grid()
plt.show()

File: series2/series2_code/series2_code/ssprk3_timestep/plot_convergence.m
clc
close all
clear all

FolderName = 'build';
error = load([FolderName,'/error.txt']);
dt = load([FolderName,'/dt.txt']);

N = length(error);
rate = polyfit(log(dt(2:N)), log(error(2:N)), 1);
rate = rate(1)

loglog(dt, error, '-o')
hold on
loglog(dt, error(N-1)*dt.^rate/(dt(N-1).^rate), '--')
xlabel('\Delta t')
ylabel('Error')
axis([dt(N) dt(1) -inf inf]);
label = strcat('\Delta t^{', num2str(rate,3), '}');
legend('|u(T)-U_N|', label, 'Location', 'northwest')
grid on
